### ISSUE TITLE
add option to enable Scroll Forward like on cmd

### ISSUE THREAD

**Issue by rafalou38** [microsoft/terminal#8623]
# Description of the new feature/enhancement

it would be nice to have an option that would allow scrolling under the end of the text so everything is not stacked at the bottom and I don't have to do `clear` every time

there is no need to be able to scroll indefinitely,  just a bit so we can see a bit better

## expected:
![image](https://user-images.githubusercontent.com/58974586/102687783-f64b8400-41f1-11eb-9b9a-f5940dbc11a6.png)
## actual:
![image](https://user-images.githubusercontent.com/58974586/102687807-1bd88d80-41f2-11eb-895f-fb64cbb4c2d3.png)

# Proposed technical implementation details

an option in the settings.json like

`"ScrollForward": true,`

**Comment by zadjii-msft** [comment]
You know what, in the past I absolutely hated this idea. It is one of those things that made it harder for the console to achieve parity with linux terminals. 

Now I'm thinking that it might be okay in a limited fashion. There'd still only be one mutable viewport, but in scrollForward mode, we'd let you scroll the viewport down (height-1) lines from the bottom line. I think everything in the Terminal's VT implementation already uses the mutable viewport's corrdinates, rather than the visible viewport, so this might be okay. 

The tricky case would be scrolling forward at the bottom of the buffer - there'd be no extra lines there, so we'd have to add them as the user tried to scroll forward. That'd be weird.

**Comment by noheromen** [comment]
Any dvelepoment done in this area? It would be really great if it was implemented. Same way as is in actuall powershell. Thanks

**Comment by zadjii-msft** [comment]
Nope. We'll make sure to update this thread when there is. In the meantime, might I recommend the Subscribe button?
![image](https://user-images.githubusercontent.com/18356694/91237459-5cbb0c80-e700-11ea-9347-b9b1ec2813b1.png)
That way you'll be notified of any updates to this thread, without needlessly pinging everyone on this thread ☺️

**Comment by noheromen** [comment]
Any progression on this task? it would be really helpful to have this option. Thanks

**Comment by zadjii-msft** [comment]
Nope. If you/someone would like to help contribute a fix here, I'd be happy to start pointing you in the right direction. But I (honestly) don't think we'll be able to get to this for a very long time. 

> [!Note]
> ## Walkthrough

(_in progress_)

* `_scrollOffset` in `Terminal.cpp`/`hpp` is what controls the distance up from the mutable viewport the user has scrolled. So scroll one line up, and the value is 1. This should be changed to also allow negative values. 
* We shouldn't allow scroll-forward in the alt buffer (for things like `vim`)
* Tricky edge case: how do we deal with scroll-forward when the buffer is full? There's literally not lines in the buffer to render at that point. We'd probably have to somehow fake it in the `RenderData`, or just prevent scroll-forward in that case entirely
* Once that all works, create a profile setting in `MTSMSettings` and plumb that through to the `TermControl`

**Comment by lukexyz** [comment]
> Any progression on this task? it would be really helpful to have this option. Thanks
I think this is a perfectly reasonable nudge. Having said that thank you for your candid response @zadjii-msft
